### PR TITLE
pub get offline for scenario_app

### PR DESCRIPTION
**Pull request by dnfield**
Part of https://github.com/flutter/flutter/issues/84290

This will make pub get run as part of `gclient sync` for these instead of as part of the build. Local analysis can still work by uncommenting the line in pubspec.yaml and re-running pub get.

### CODE REVIEW

**Comment by dnfield** [comment]
Ahhh this is making analysis fail. I'll have to think about if we can get both to work... My preference would be to skip analysis though if we can't have both.

**Comment by dnfield** [comment]
Ok. Updated this patch so that `_embedder.yaml` is valid for in tree usage, and the `_embedder.yaml` file for out of tree usage is generated at build time with the right paths. That way, in tree can safely refer to the in tree version of sky_engine without worrying about build. This is passing locally, let's see what CI thinks.

I'm also updating testing/dart to use this pattern.

File: sky/packages/sky_engine/BUILD.gn [hunk 1]
```diff
@@ -181,16 +181,45 @@ group("copy_dart_sdk") {
   ]
 }
 
+generated_file("_embedder_yaml") {
+  outputs = [ "$root_gen_dir/dart-pkg/sky_engine/lib/_embedder.yaml" ]
+  contents = [
+    "embedded_libs:",
```
Comment (zanderso) [inline sky/packages/sky_engine/BUILD.gn:187]: Please include comments that say that the file is generated, not to modify it by hand, and how to generate it.

File: sky/packages/sky_engine/lib/_embedder.yaml [hunk 2]
```diff
@@ -1,23 +1,25 @@
+# This file is suitable for use within the tree. A different _embedder.yaml
+# is generated in BUILD.gn. Changes here must be mirrored there.
```
Comment (zanderso) [inline sky/packages/sky_engine/lib/_embedder.yaml:2]:
```suggestion
# This file is suitable for use within the tree. A different _embedder.yaml
# is generated by the BUILD.gn in this directory. Changes here must be
# mirrored there.
```

**Review by zanderso — APPROVED** [review]

File: sky/packages/sky_engine/BUILD.gn (hunk 1 above)
Comment (dnfield) [reply sky/packages/sky_engine/BUILD.gn:190]: Done

**Comment by dnfield** [comment]
This was really part of https://github.com/flutter/flutter/issues/84618

**Comment by dnfield** [comment]
And https://github.com/flutter/flutter/issues/53455

**Pull request merged**